### PR TITLE
Add block PV research doc

### PR DESCRIPTION
**Pull request by ivan4th**
This change is [Reviewable](https://reviewable.io/reviews/mirantis/virtlet/692)

### CODE REVIEW

**Comment by jellonek** [comment]
Reviewed 1 of 1 files at r1.
Review status: 0 of 2 LGTMs obtained

---

*[docs/design-proposals/pv.md, line 4 at r1](https://reviewable.io/reviews/mirantis/virtlet/692#-LEtFgQj8G32B5lvBSs5:-LEtFgQj8G32B5lvBSs6:b-woekp4) ([raw file](https://github.com/mirantis/virtlet/blob/f236d1f9a75957865b47b1842059cee75ca89ee5/docs/design-proposals/pv.md#L4)):*
> ```Markdown
> 
> Virtlet uses custom flexvolume driver to handle raw block devices and
> Ceph volumes right now. Another problem is that we may want to support
> ```

First sentence is not described as problem, so "Another" there is not in place :)

---

*[docs/design-proposals/pv.md, line 35 at r1](https://reviewable.io/reviews/mirantis/virtlet/692#-LEtJK8tAd1EsTqMP1iB:-LEtJK8tAd1EsTqMP1iC:beu1uqk) ([raw file](https://github.com/mirantis/virtlet/blob/f236d1f9a75957865b47b1842059cee75ca89ee5/docs/design-proposals/pv.md#L35)):*
> ```Markdown
>   devices:
>   - container_path: /dev/testpvc
>     host_path: /var/lib/kubelet/pods/65b0c985-6e81-11e8-be27-769e6e14e66a/volumeDevices/kubernetes.io~local-volume/local-block-pv
> ```

Is that a path to link, which points to some device, or it can be device node created with `mknod`?
Can that be directly consumed by libvirt?
How that can be problematic with libvirt acls which we are configuring in https://github.com/Mirantis/virtlet/blob/master/images/image_skel/etc/libvirt/qemu.conf#L8 ?

---


*Comments from [Reviewable](https://reviewable.io/reviews/mirantis/virtlet/692)*

**Comment by ivan4th** [comment]
Review status: 0 of 2 LGTMs obtained

---

*[docs/design-proposals/pv.md, line 4 at r1](https://reviewable.io/reviews/mirantis/virtlet/692#-LEtFgQj8G32B5lvBSs5:-LFGwQQa36RL7TE-9Phv:bj3eoea) ([raw file](https://github.com/mirantis/virtlet/blob/f236d1f9a75957865b47b1842059cee75ca89ee5/docs/design-proposals/pv.md#L4)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

First sentence is not described as problem, so "Another" there is not in place :)
</blockquote></details>

Fixed, thanks.

---

*[docs/design-proposals/pv.md, line 35 at r1](https://reviewable.io/reviews/mirantis/virtlet/692#-LEtJK8tAd1EsTqMP1iB:-LFGw_eR3RniPE-D_4RF:bo8p0vq) ([raw file](https://github.com/mirantis/virtlet/blob/f236d1f9a75957865b47b1842059cee75ca89ee5/docs/design-proposals/pv.md#L35)):*
<details><summary><i>Previously, jellonek (Piotr Skamruk) wrote…</i></summary><blockquote>

Is that a path to link, which points to some device, or it can be device node created with `mknod`?
Can that be directly consumed by libvirt?
How that can be problematic with libvirt acls which we are configuring in https://github.com/Mirantis/virtlet/blob/master/images/image_skel/etc/libvirt/qemu.conf#L8 ?
</blockquote></details>

See the "experimental" part of this document, I show that it's a symlink to `/dev/...` there.

---


*Comments from [Reviewable](https://reviewable.io/reviews/mirantis/virtlet/692)*

**Comment by jellonek** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: 0 of 2 LGTMs obtained, and 1 stale

---



*Comments from [Reviewable](https://reviewable.io/reviews/mirantis/virtlet/692#-:-LFGxP43D15BmZndC-2Q:bnfp4nl)*

File: docs/design-proposals/pv.md [hunk 1]
```diff
@@ -0,0 +1,660 @@
...
+original one is destroyed. For local PVs, the scheduler will always
+place the pod on the node where the local PV resides (this constitutes
+so called "gravity"). There's a problem with this approach, namely,
+there's no reliable way for a CRI implementation to find a PV that
+corresponds to a block device, so Virtlet will have to examine the
+contents of the PV to see if it's used for the first time. This also
+means that Virtlet will have hard time establishing the correspondence
+between PVs and the images that are applied to them (e.g. imagine a PV
+being used by a pod with different image later). It's possible to
+overcome these problems by either storing the metadata on the block
```
Comment (fabiand) [inline docs/design-proposals/pv.md]: You might want to look at CDI here and DataVolumes which came up in KubeVirt to solve the provisioning problem

**Pull request merged**